src/ui/zcl_abapgit_password_dialog.clas.abap

```diff
@@ -22,10 +22,39 @@ CLASS ZCL_ABAPGIT_PASSWORD_DIALOG IMPLEMENTATION.
 
   METHOD popup.
 
-    PERFORM password_popup
-      IN PROGRAM (sy-cprog)
-      USING iv_repo_url
-      CHANGING cv_user cv_pass.
+    DATA lv_gui_is_available TYPE abap_bool.
+
+    lv_gui_is_available = zcl_abapgit_ui_factory=>get_gui_functions( )->gui_is_available( ).
+
+    IF lv_gui_is_available = abap_true.
+      PERFORM password_popup
+        IN PROGRAM (sy-cprog)
+        USING iv_repo_url
+        CHANGING cv_user cv_pass.
+    ELSE.
+      "check if ZCL_ABAPGIT_DEFAULT_AUTH_INFO exists (part of ADT_Backend project)
+      TRY.
```

Comment:
suggest adding a comment in the code that this is a temporary workaround, with a link to this PR